rustls/src/time_provider.rs

```diff
@@ -0,0 +1,29 @@
+//! no-std time provider
+
+use core::fmt::Debug;
+
+use pki_types::UnixTime;
+
+/// An object that provides the current time.
+///
+/// This is used to, for example, check if a certificate has expired during certificate validation,
+/// or to check the age of a ticket.
+#[allow(unreachable_pub)]
```

Comment:
Not sure this `unreachable_pub` makes sense, we should allow access to this before this PR is merged, right -- so might as well do it in this commit?

Reply:
(Maybe squash this commit with the first one that actually introduces usage of the trait?)

Reply:
Fixed.